src/__tests__/tree.spec.ts

```diff
@@ -985,4 +985,17 @@ describe('SchemaTree', () => {
       });
     });
   });
+
+  describe('recursive walking', () => {
+    it('should load with a max depth', async () => {
+      const schema = JSON.parse(
+        await fs.promises.readFile(path.resolve(__dirname, '__fixtures__', 'recursive-schema.json'), 'utf8'),
+      );
+
+      const w = new SchemaTree(schema, {
+        maxRefDepth: 1000,
```

Comment:
Just a note that after the fix processing this schema will work properly and won't reach max depth